docs/userguide/miscellaneous.rst

```diff
@@ -94,3 +94,38 @@ correctly when installed as a zipfile, correct any problems if you can, and
 then make an explicit declaration of ``True`` or ``False`` for the ``zip_safe``
 flag, so that it will not be necessary for ``bdist_egg`` to try to guess
 whether your project can work as a zipfile.
+
+
+.. _Controlling files in the distribution:
+
+Controlling files in the distribution
+-------------------------------------
+
+For the most common use cases, ``setuptools`` will automatically 
+find out which files are necessary for distributing the package.
+This includes all pure Python modules in the ``py_modules`` or ``packages``
+configuration and all C sources listed as part of extensions
+(it doesn't catch C headers, though).
+
+However, when building more complex packages (e.g. packages that include
```

Comment:
I think it is worth calling out that wheels (and the final installed package) contain only the data files that are contained *within* listed `package` directories, *and* only when `include_package_data` is True.  Specifically this can be used to include a top-level `tests` folder in an `sdist` using `MANIFEST.in`.  But by not including `tests` in the `packages` metadata (by using explicit packages or `include`/`exclude` with `find_packages`, you avoid shipping the tests in the wheel and polluting `site-packages` with a `tests` folder.  The tests in the sdist can be used by downstream packagers like Linux distros or conda-forge to test the package before re-packaging.  Maybe this should go in the PyPUG itself?

Reply:
>  I think it is worth calling out that wheels (and the final installed package) contain only the data files that are contained within listed `package` directories, *and* only when `include_package_data` is True

Ummm... in theory wheels *could* contain files outside the `package` directories via the [`.data`](https://packaging.python.org/en/latest/specifications/binary-distribution-format/#the-data-directory) subdir in the wheel specification. That however is super tricky, because it will depend on the installation schema that varies with OS/platform/virtualenv etc...
So we want to move users away from that.

> Maybe this should go in the PyPUG itself?

It is definitely a good idea to add a discussion about tests/docs in the sdist to PyPUB! Maybe at some point in the future I will give it a try and propose a PR there (but please also feel free to go ahead and open one if you have the time).

Reply:
Hi @blink1073, in [8252975](https://github.com/pypa/setuptools/pull/3148/commits/82529755d433de93fcce1c48385dce3cf4003253) I tried to make these relationships more clear. Would you mind to have a look and see if the text is more clear/has improved?

There is also a link to the rendered version here: https://setuptools--3148.org.readthedocs.build/en/3148/userguide/miscellaneous.html#controlling-files-in-the-distribution